x/wasm/internal/keeper/handler_plugin.go

```diff
@@ -121,12 +121,12 @@ func EncodeStakingMsg(sender sdk.AccAddress, msg *wasmTypes.StakingMsg) ([]sdk.M
 		if err != nil {
 			return nil, err
 		}
-		sdkMsg := staking.MsgDelegate{
-			DelegatorAddress: sender,
-			ValidatorAddress: validator,
+		sdkMsg := stakingtypes.MsgDelegate{
+			DelegatorAddress: sender.String(),
+			ValidatorAddress: validator.String(),
```

Comment:
We got the bech32 from the contract. On 116 we convert to binary, and here we convert back to bech32.

Why not delete 116-119 and replace this line with `ValidatorAddress: msg.Delegate.Validator`? Cleaner.